src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1560,12 +1560,26 @@ def domain_unbind(cmd, client, resource_group, service, app, domain_name):
     return client.custom_domains.delete(resource_group, service, app, domain_name)
 
 
-def get_app_insights_key(cli_ctx, resource_group, name):
+def get_conenction_string_or_instrumentation_key(appinsights):
+    credential = None
+    if appinsights:
+        if appinsights.conenction_string:
+            credential = appinsights.connection_string
+        elif appinsights.instrumentation_key:
+            credential = appinsights.instrumentation_key
+    return credential
+
+
+def get_app_insights_credential(cli_ctx, resource_group, name):
+    """
+    Application Insights is migrating from instrumentation_key to connection_key.
+    Get connection_string first, if not exist, fall back to instrumentation_key.
+    """
     appinsights_client = get_mgmt_service_client(cli_ctx, ApplicationInsightsManagementClient)
     appinsights = appinsights_client.components.get(resource_group, name)
-    if appinsights is None or appinsights.instrumentation_key is None:
+    if appinsights is None or (appinsights.connection_string is None and appinsights.instrumentation_key is None):
```

Comment:
Pre-condition, upgrade the dependency from ```0.1.1``` to ```1.0.0``` within ```azure-cli``` for ```azure-mgmt-applicationinsights```, otherwise, ```connection_string``` is not available from the ```Application Insights``` data model.

Reply:
OK, let me label this PR as `do-not-merge` at first~

Reply:
We can see from below link that the version is at ```1.0.0``` now:
https://github.com/Azure/azure-cli/blob/master/src/azure-cli/requirements.py3.windows.txt#L24